packaging/linux/post_install.sh

```diff
@@ -173,6 +173,14 @@ fix_bad_config_perms() {
 
 fix_bad_config_perms
 
+# Associate Keybase with .saltpack files.
+if command -v xdg-mime &> /dev/null ; then
+  # Associate .saltpack with application/x-saltpack.
+  xdg-mime install --mode system /opt/keybase/x-saltpack.xml
+  # Associate Keybase app with application/x-saltpack.
+  xdg-mime default keybase.desktop application/x-saltpack
```

Comment:
Run as root, this will modify /root/.config/mimeapps.list

And it cannot be run as the user.

The default only matters if multiple desktop files register themselves to the same mimetype, and you want to specify which one to use (the reference implementation will, when no default is added, fall back to randomly choosing an opener).

There are already mimetypes registered in keybase.desktop, but you don't `xdg-mime default` them. This needs to be removed from post_install.sh and added to the desktop file directly.

Reply:
Note that /usr/share/applications/mimeinfo.cache which contains the quick lookup cache for this can be regenerated using `update-desktop-database`, following the same logic as `gtk-update-icon-cache`.

This was already not happening in post_install.sh for the existing mimetypes (but distros such as archlinux where the package manager handles `gtk-update-icon-cache` automatically, also handle `update-desktop-database` automatically).